agent/consul/server.go

```diff
@@ -816,6 +816,7 @@ func newGRPCHandlerFromConfig(deps Deps, config *Config, s *Server) connHandler
 
 		// Note: these external gRPC services are also exposed on the internal server to
 		// enable RPC forwarding.
+		s.peerStreamServer.Register(srv)
```

Comment:
Can you add a test to ensure that forwarding is working? Here's an example:
https://github.com/hashicorp/consul-enterprise/blob/main/agent/consul/partition_backend_test.go#L84

Reply:
Added in [5078d6e](https://github.com/hashicorp/consul/pull/14119/commits/5078d6e0a25ee8bd3ffc8f792b10aac49752a913)

Reply:
Confirmed `unknown service` comes up without this change